crates/oxc_ast_visit/src/utf8_to_utf16/translation.rs

```diff
@@ -150,10 +259,16 @@ pub fn build_translations(source_text: &str, translations: &mut Vec<Translation>
         // `ptr < body_end_ptr` check ensures it's valid to read `CHUNK_SIZE` bytes starting at `ptr`.
         #[expect(clippy::cast_ptr_alignment)]
         let chunk = unsafe { ptr.cast::<AlignedChunk>().as_ref().unwrap_unchecked() };
-        if chunk.contains_unicode() {
-            // SAFETY: `ptr` is equal to or after `start_ptr`. Both are within bounds of `bytes`.
-            // `ptr` is derived from `start_ptr`.
-            let offset = unsafe { ptr.offset_from_unsigned(start_ptr) };
+
+        // SAFETY: `ptr` is equal to or after `start_ptr`. Both are within bounds of `bytes`.
+        // `ptr` is derived from `start_ptr`.
+        let offset = unsafe { ptr.offset_from_unsigned(start_ptr) };
```

Comment:
Move this into the `if has_unicode || has_line_breaks { ... }` block. If no line breaks or unicode chars in the block, we don't use this value, so avoid calculating it in that case.